jlexer/lexer.go

```diff
@@ -425,13 +425,19 @@ func (r *Lexer) errInvalidToken(expected string) {
 	}
 }
 
+func (r *Lexer) errInvalidToken2(c byte) {
+	if r.err == nil {
+		r.errInvalidToken(string([]byte{c}))
+	}
+}
+
 // Delim consumes a token and verifies that it is the given delimiter.
 func (r *Lexer) Delim(c byte) {
 	if r.token.kind == tokenUndef && r.Ok() {
 		r.fetchToken()
 	}
 	if !r.Ok() || r.token.delimValue != c {
```

Comment:
What I'm saying is that there is already an "if" here. You could simply add an extra check of `err != nil` to avoid []byte->string conversion (and calling the function) if there is already an error.

It would work pretty much the same way and I would prefer this as it will add almost no extra code.

Alternatively, you could come up with a name better than `errInvalidToken2`. The problem with `errInvalidToken2` is that it's not a very readable name.